src/sentry/sentry_metrics/consumers/indexer/parallel.py

```diff
@@ -152,6 +154,14 @@ def __init__(
             initializer=functools.partial(initialize_subprocess_state, self.config),
         )
 
+        if use_case is UseCaseKey.PERFORMANCE and options.get(
```

Comment:
If this works I assume we would get rid of this condition on the use case

Reply:
Also I'd assume you added the delay in the factory initialization as a quick way to test if the approach generates an improvement. If this works well we should properly integrate it in the consumer startup and not hide it in the factory startup. The factory is not guaranteed to be only created at consumer startup while today it does. Relying on this behavior is adding a silent assumption about the design of arroyo which is not really true and which we do not need.

Reply:
Yeah, I think this makes sense. There's some context here https://github.com/getsentry/sentry/pull/62018#discussion_r1433103851 but basically there was a suggestion to move it to the factory, I believe in order to hide this implementation detail. 

But yes once we validate this we can move it to make sure we aren't relying on any Arroyo design assumptions.